File: data/derived_cot/rq3_prompting/results/translation/qwen/java_to_cpp/code_output/JSONProcessor.cpp
#include <nlohmann/json.hpp>
#include <fstream>
#include <filesystem>
#include <string>

namespace fs = std::filesystem;

class JSONProcessor {
public:
    nlohmann::json readJson(const std::string& filePath) {
        if (!fs::exists(filePath)) {
            return nlohmann::json(nullptr);
        }
        try {
            std::ifstream i(filePath);
            return nlohmann::json::parse(i);
        } catch (...) {
            return nlohmann::json(nullptr);
        }
    }

    bool writeJson(const nlohmann::json& data, const std::string& filePath) {
        try {
            std::ofstream o(filePath);
            o << data.dump(0); // 0 for no indentation
            return true;
        } catch (...) {
            return false;
        }
    }

    bool processJson(const std::string& filePath, const std::string& removeKey) {
        nlohmann::json data = readJson(filePath);
        if (data.is_null()) {
            return false;
        }
        if (data.contains(removeKey)) {
            data.erase(removeKey);
            return writeJson(data, filePath);
        }
        return false;
    }
};

File: data/derived_cot/rq1_traces/translation/qwen/java_to_py/code_output/JSONProcessor.py
import json
import os

class JSONProcessor:

    @staticmethod
    def read_json(filePath):
        if not os.path.exists(filePath):
            return None
        try:
            with open(filePath, 'r') as file:
                return json.load(file)
        except Exception:
            return None

    @staticmethod
    def write_json(data, filePath):
        try:
            with open(filePath, 'w') as file:
                json.dump(data, file, indent=None)
            return True
        except Exception:
            return False

    @staticmethod
    def process_json(filePath, removeKey):
        data = JSONProcessor.read_json(filePath)
        if data is None:
            return False
        if removeKey in data:
            del data[removeKey]
            return JSONProcessor.write_json(data, filePath)
        return False